src/pages/Account/Tabs/ModulesTab.tsx

```diff
@@ -55,19 +83,169 @@ function ModulesContent({
   );
 }
 
-type ModulesTabProps = {
+function ModulesContentReworked({address}: {address: string}): JSX.Element {
+  const {data: registry} = useGetAccountResource(
+    address,
+    "0x1::code::PackageRegistry",
+  );
+
+  const [selectedModuleIndex, setSelectedModuleIndex] =
+    React.useState<number>(0);
+
+  const packages: PackageMetadata[] =
+    registry === undefined ? [] : (registry.data as any).packages;
+  const modules = orderBy(
+    packages.flatMap((p) => p.modules),
+    "name",
+  );
+
+  if (modules.length === 0) {
+    return <EmptyTabContent />;
+  }
+
+  return (
+    <Grid container spacing={2}>
+      <Grid
+        item
+        md={3}
+        sx={{
+          maxHeight: "500px",
+          overflowY: "auto",
+        }}
+      >
+        <ModuleSidebar
+          moduleNames={modules.map((m) => m.name)}
+          selectedModuleIndex={selectedModuleIndex}
+          setSelectedModuleIndex={setSelectedModuleIndex}
+        />
+      </Grid>
+      <Grid item md={9}>
+        <ModuleContent
+          address={address}
+          moduleName={modules[selectedModuleIndex].name}
+          sourceCode={modules[selectedModuleIndex].source}
+        />
+      </Grid>
+    </Grid>
+  );
+}
+
+function ModuleSidebar({
+  moduleNames,
+  selectedModuleIndex,
+  setSelectedModuleIndex,
+}: {
+  moduleNames: string[];
+  selectedModuleIndex: number;
+  setSelectedModuleIndex: (index: number) => void;
+}) {
+  return (
+    <Box sx={{padding: "24px"}}>
+      <Typography fontSize={16} fontWeight={500} marginBottom={"24px"}>
+        Modules
+      </Typography>
+      {moduleNames.map((moduleName, i) => (
+        <ModuleOption
+          key={i}
+          onClick={() => setSelectedModuleIndex(i)}
+          selected={i === selectedModuleIndex}
+          name={moduleName}
+        />
+      ))}
+    </Box>
+  );
+}
+
+function ModuleOption({
+  onClick,
+  selected,
+  name,
+}: {
+  onClick: () => void;
+  selected: boolean;
+  name: string;
+}) {
+  const theme = useTheme();
+
+  return (
+    <Box
+      key={name}
+      onClick={onClick}
+      sx={{
+        fontSize: 12,
+        fontWeight: 500,
+        padding: "8px",
+        bgcolor: !selected
+          ? "transparent"
+          : theme.palette.mode === "dark"
+          ? grey[500]
+          : grey[200],
+        ":hover": {
+          cursor: "pointer",
+        },
+      }}
+    >
+      {name}
+    </Box>
+  );
+}
+
+function ModuleContent({
+  address,
+  moduleName,
+  sourceCode,
+}: {
   address: string;
-  accountData: Types.AccountData | undefined;
-};
+  moduleName: string;
+  sourceCode: string;
+}) {
+  React.useEffect(() => {
+    hljs.highlightAll();
```

Comment:
highlightjs is not working with the dark mode atm . would this [one](https://github.com/react-syntax-highlighter/react-syntax-highlighter#readme) work better?
<img width="1534" alt="Screenshot 2023-03-02 at 10 36 21 AM" src="https://user-images.githubusercontent.com/121921928/222520818-bfe9fea1-b873-48b7-9897-b74fc3dccee7.png">

Reply:
also it seems like the sidebar scroll position is off?

Reply:
how do you want the sidebar scroll position to behave? i am not too sure

Reply:
> also it seems like the sidebar scroll position is off?

how do you want the sidebar scroll position to behave? i am not too sure

Reply:
for sidebar scroll, i meant if you look at the screenshot, the scroll bar is all the way in the "Coins" tab.

Reply:
> for sidebar scroll, i meant if you look at the screenshot, the scroll bar is all the way in the "Coins" tab.

so should I move it to a lefter position? @jillxuu

Reply:
I should make it more clear. if you look at this casting, the scroll bar's position is off, meaning that top of the scroll exceeds the box limit and gets into the "Coins" tab. not sure if it's only an issue on netlify though. 

https://user-images.githubusercontent.com/121921928/222588942-428d1a4f-38eb-4a45-8a5a-cacbf43fea33.mov

Reply:
irt the color scheme, i tried sth out in my dev env and it should work:

```
{theme.palette.mode === "dark" ? (
      <link
        rel="stylesheet"
        href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/11.7.0/styles/monokai-sublime.min.css"
      />
    ) : (
      <link
        rel="stylesheet"
        href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/11.7.0/styles/base16/tomorrow.min.css"
      />
    )}
    <pre>
      <code className="language-rust">{transformCode(sourceCode)}</code>
    </pre>
```